tests/api2/test_440_snmp.py

```diff
@@ -63,65 +132,254 @@ def validate_snmp_get_sysname_uses_same_ip(hostip):
     assert get_dst.endswith(".161")
 
 
-def test_01_Configure_SNMP():
-    results = PUT('/snmp/', {
+def user_list_users(snmp_config):
+    """Run an snmpwalk as a SNMP v3 user"""
+
+    add_cmd = None
+    if snmp_config['v3_privproto']:
+        authpriv_setting = 'authPriv'
+        add_cmd = f"-x {snmp_config['v3_privproto']} -X \"{snmp_config['v3_privpassphrase']}\" "
+    else:
+        authpriv_setting = 'authNoPriv'
+
+    cmd = f"snmpwalk -v3 -u  {snmp_config['v3_username']} -l {authpriv_setting} "
+    cmd += f"-a {snmp_config['v3_authtype']} -A {snmp_config['v3_password']} "
+    if add_cmd:
+        cmd += add_cmd
+    cmd += "localhost iso.3.6.1.6.3.15.1.2.2.1.3"
+
+    # This call will timeout if SNMP is not running
+    res = ssh(cmd)
+    return [x.split()[-1].strip('\"') for x in res.splitlines()]
+
+
+@pytest.mark.dependency(name='SNMP_CONFIGURED', scope="module")
+def test_01_Configure_SNMP(initialize_for_snmp_tests):
+    config = initialize_for_snmp_tests
+
+    # We should be starting with the default config
+    # Check the hard way so that we can identify the culprit
+    for k, v in EXPECTED_DEFAULT_CONFIG.items():
+        assert config.get(k) == v, f'Expected {k}:"{v}", but found {k}:"{config.get(k)}"'
+
+    # Make some changes that will be checked in a later test
+    call('snmp.update', {
         'community': COMMUNITY,
         'traps': TRAPS,
         'contact': CONTACT,
-        'location': LOCATION,
-        'v3_password': PASSWORD})
-    assert results.status_code == 200, results.text
-
+        'location': LOCATION
+    })
 
-def test_02_Enable_SNMP_service_at_boot():
-    results = PUT('/service/id/snmp/', {'enable': True})
-    assert results.status_code == 200, results.text
 
+@pytest.mark.dependency(name='SNMP_ENABLED', scope="module")
+def test_02_Enable_SNMP_service_at_boot(request):
+    depends(request, ["SNMP_CONFIGURED"], scope="module")
+    id = call('service.update', 'snmp', {'enable': True})
+    assert isinstance(id, int)
 
-def test_03_verify_snmp_do_not_leak_password_in_middleware_log(request):
-    cmd = f"""grep -R "{PASSWORD}" /var/log/middlewared.log"""
-    results = SSH_TEST(cmd, user, password)
-    assert results['result'] is False, str(results['output'])
 
+def test_03_verify_snmp_does_not_leak_password_in_logs():
+    with pytest.raises(AssertionError):
+        ssh(f'grep -R "{PASSWORD}" /var/log/middlewared.log')
```

Comment:
I don't think this is a meaningful test (I can't see an avenue for it to leak and we have separate tests to validate auto-redaction of fields marked as `private=True`